### PR TITLE
add org.speed_dreams.SpeedDreams

### PR DESCRIPTION
**Pull request by son-link**

### CODE REVIEW

File: org.speed.dreams.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  - '*.a'
+  - '*.la'
+  - '/share/speed-dreams-2/cmake/'
+  
+modules:
+  - shared-modules/glu/glu-9.0.0.json
+
+  - name: giflib
+    config-opts:
+      - --prefix=/app
```
Comment (bilelmoussaoui) [inline org.speed.dreams.yaml]: Not needed. It's set by default now

File: org.speed.dreams.yaml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      - --disable-static
+    sources:
+      - type: archive
+        url: http://downloads.sourceforge.net/sourceforge/giflib/giflib-5.1.4.tar.bz2
+        sha256: 2c171ced93c0e83bb09e6ccad8e3ba2b
+  
+  - name: freeglut
+    buildsystem: cmake-ninja
+    config-opts:
+    - -DCMAKE_INSTALL_LIBDIR:PATH=/app/lib
```
Comment (bilelmoussaoui) [inline org.speed.dreams.yaml]: I think it's not needed anymore neither

File: org.speed.dreams.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    
+    sources:
+      - type: archive
+        url: https://downloads.sourceforge.net/freeglut/freeglut-3.0.0.tar.gz
+        sha256: fca52242f9344627a30f11487ee42002e6b0dacd
+        
+  - name: jasper
+    buildsystem: cmake-ninja
+    config-opts:
+    - -DCMAKE_INSTALL_PREFIX=/app
```
Comment (bilelmoussaoui) [inline org.speed.dreams.yaml]: Not needed

File: org.speed.dreams.yaml [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+      - type: shell
+        commands:
+          - tar xf speed-dreams-src-base-2.2.2-rc2-r6527.tar.xz
+          - tar xf speed-dreams-src-wip-cars-and-tracks-2.2.2-rc2-r6527.tar.xz
+          - tar xf speed-dreams-src-hq-cars-and-tracks-2.2.2-rc2-r6527.tar.xz
+          - tar xf speed-dreams-src-more-hq-cars-and-tracks-2.2.2-rc2-r6527.tar.xz
+          
+    post-install:
+      - mkdir -p /app/share/appdata
```
Comment (bilelmoussaoui) [inline org.speed.dreams.yaml]: As you're using install for the appdata file, there's no need to use mkdir here. You could do the same for the icons & the desktop file

File: org.speed.dreams.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">http://speed-dreams.org/images/sd-20/speed-dreams-20-car-supercars.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-daytime-dusk.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-car-ls1.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-track-garguree.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-daytime-night.jpg</screenshot>
+  </screenshots>
+  <url type="homepage">http://speed-dreams.org/</url>
```
Comment (bilelmoussaoui) [inline org.speed.dreams.appdata.xml]: Lacks OARS https://hughsie.github.io/oars/

File: org.speed.dreams.yaml (hunk 4 above)
Comment (bilelmoussaoui) [reply org.speed.dreams.yaml]: I meant the `install` command. Using `install -D` will create the `DEST` folder too.

Comment (son-link) [reply org.speed.dreams.yaml]: Aaa, OK

**Comment by nedrichards** [comment]
bot, build org.speed_dreams.SpeedDreams

File: org.speed.dreams.yaml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+runtime-version: '18.08'
+sdk: org.freedesktop.Sdk
+command: speed-dreams-2
+
+finish-args: 
+  - --socket=wayland
+  - --share=ipc
+  - --socket=x11
+  - --device=all
+  - --filesystem=~/.speed-dreams-2
```
Comment (barthalion) [inline org.speed.dreams.yaml]: Use `--persist=.speed-dreams-2` instead.

File: org.speed.dreams.yaml [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DSD_BINDIR=/app/bin
+      - -DSD_DATADIR=/app/share/speed-dreams-2
+      - -DSD_LIBDIR=/app/lib/speed-dreams-2
+      - -DOPTION_OFFICIAL_ONLY=ON
+    
+    sources:
+      - type: file
+        url: https://downloads.sourceforge.net/sourceforge/speed-dreams/speed-dreams-src-base-2.2.2-rc2-r6527.tar.xz
+        md5: 956bda0acbcb89eaa2ca742dbedc8941
```
Comment (barthalion) [inline org.speed.dreams.yaml]: Use sha256 for checksums. It's de facto standard on Flathub.

File: org.speed.dreams.yaml [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        
+      - type: file
+        path: org.speed_dreams.SpeedDreams.desktop
+        
+      - type: file
+        path: org.speed_dreams.SpeedDreams.appdata.xml
+
+      - type: shell
+        commands:
+          - tar xf speed-dreams-src-base-2.2.2-rc2-r6527.tar.xz
```
Comment (barthalion) [inline org.speed.dreams.yaml]: All source types take `dest` argument. Why not use it instead?

File: org.speed_dreams.SpeedDreams.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-car-ls1.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-track-garguree.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-daytime-night.jpg</screenshot>
+  </screenshots>
+  <url type="homepage">http://speed-dreams.org/</url>
+  <updatecontact>speed-dreams-devel@lists.sourceforge.net</updatecontact>
+  <releases>
+    <release version="2.2.2-rc2" date="2018-03-09"/>
+  </releases>
+</application>
```
Comment (barthalion) [inline org.speed_dreams.SpeedDreams.appdata.xml]: File misses OARS; should be entirely contributed upstream.

File: org.speed_dreams.SpeedDreams.desktop [hunk 10]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (barthalion) [inline org.speed_dreams.SpeedDreams.desktop]: Should be contributed upstream.

**Comment by barthalion** [comment]
Have you contacted upstream developers about your pull request?

**Comment by leillo1975** [comment]
Devs contacted:

https://sourceforge.net/p/speed-dreams/mailman/message/36429876/

File: org.speed.dreams.yaml [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    config-opts:
+      - --disable-static
+    sources:
+      - type: archive
+        url: http://downloads.sourceforge.net/sourceforge/giflib/giflib-5.1.4.tar.bz2
+        sha256: 2c171ced93c0e83bb09e6ccad8e3ba2b
+  
+  - name: freeglut
+    buildsystem: cmake-ninja
+
```
Comment (TingPing) [inline org.speed.dreams.yaml]: Remove extra newline

File: org.speed.dreams.yaml [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    
+    sources:
+      - type: archive
+        url: https://downloads.sourceforge.net/freeglut/freeglut-3.0.0.tar.gz
+        sha256: fca52242f9344627a30f11487ee42002e6b0dacd
+        
+  - name: jasper
+    buildsystem: cmake-ninja
+    config-opts:
+    - -DCMAKE_INSTALL_LIBDIR=lib
```
Comment (TingPing) [inline org.speed.dreams.yaml]: *shouldn't* be needed.

File: org.speed.dreams.yaml [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+  
+  - name: libxmu
+    sources:
+      - type: archive
+        url: https://www.x.org/archive//individual/lib/libXmu-1.1.2.tar.bz2
+        sha256: 756edc7c383254eef8b4e1b733c3bf1dc061b523c9f9833ac7058378b8349d0b
+  
+  - name: plib
+    rm-configure: true
+    config-opts:
```
Comment (TingPing) [inline org.speed.dreams.yaml]:
```yaml
build-options:
  cxxflags: -fPIC
```

File: org.speed_dreams.SpeedDreams.appdata.xml [hunk 14]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014 MadBad <madbad82@gmail.com> -->
+<application>
+  <id type="desktop">speed-dreams-2.desktop</id>
+  <id>org.speed.dreams</id>
```
Comment (TingPing) [inline org.speed_dreams.SpeedDreams.appdata.xml]: Wrong id.

File: org.speed_dreams.SpeedDreams.appdata.xml [hunk 15]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014 MadBad <madbad82@gmail.com> -->
+<application>
```
Comment (TingPing) [inline org.speed_dreams.SpeedDreams.appdata.xml]: Run `appstream-util upgrade` on this file.

File: org.speed_dreams.SpeedDreams.appdata.xml [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+  </description>
+  <screenshots>
+    <screenshot type="default">http://speed-dreams.org/images/sd-20/speed-dreams-20-car-supercars.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-daytime-dusk.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-car-ls1.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-track-garguree.jpg</screenshot>
+    <screenshot>http://speed-dreams.org/images/sd-20/speed-dreams-20-daytime-night.jpg</screenshot>
+  </screenshots>
+  <url type="homepage">http://speed-dreams.org/</url>
+  <updatecontact>speed-dreams-devel@lists.sourceforge.net</updatecontact>
```
Comment (TingPing) [inline org.speed_dreams.SpeedDreams.appdata.xml]: `update_contact`

File: org.speed.dreams.yaml (hunk 13 above)
Comment (son-link) [reply org.speed.dreams.yaml]: > ```yaml
> build-options:
>   cxxflags: -fPIC
> ```

Don't works build-options

File: org.speed.dreams.yaml (hunk 8 above)
Comment (son-link) [reply org.speed.dreams.yaml]: I try with many dests, but don't work

**Comment by barthalion** [comment]
bot, build org.speed_dreams.SpeedDreams

File: org.speed.dreams.yaml (hunk 7 above)
Comment (son-link) [reply org.speed.dreams.yaml]: I change all checksum to sha256

**Comment by leillo1975** [comment]
What's the problem with this package?

**Comment by madbad** [comment]
Hello everyone.
I'm a member of the Speed-Dreams dev-team.
I would like to see this landing (thanks again son-link for the effort).
I'm posting just to have clear explanation of what we need to do to fill the request.
Do we just have to update our appdata file?
Should we also add the yaml file to our repo?
Also, is there some naming and placing convention we should respect?
Thanks in advance

File: org.speed_dreams.SpeedDreams.desktop [hunk 17]
```diff
@@ -0,0 +1,9 @@
...
+Version=1.0
+Type=Application
+Name=Speed Dreams
+Comment=The Open Racing Car Simulator Fork
+Icon=org.speed_dreams.SpeedDreams
+Exec=speed-dreams-2
+Categories=Game;Application;
```
Comment (AsciiWolf) [inline org.speed_dreams.SpeedDreams.desktop]: Relevant subcategory should also be added: _Application;Game;**SportsGame;**_

Comment (son-link) [reply org.speed_dreams.SpeedDreams.desktop]: Done

**Comment by TingPing** [comment]
> Hello everyone.
> I'm a member of the Speed-Dreams dev-team.
> I would like to see this landing (thanks again son-link for the effort).

Good to hear!

> Do we just have to update our appdata file?

You should pull in the desktop and appdata file used here yes (I had one comment on something to add to it).

> Should we also add the yaml file to our repo?

You do not need the flatpak manifest in tree.

> Also, is there some naming and placing convention we should respect?

All icons, desktop, and appdata files should have the `org.speed_dreams.SpeedDreams` prefix. Placed in the standard locations (`$datadir/applications`, `$datadir/metainfo`, `$datadir/icons` for example).

**Comment by TingPing** [comment]
bot, build org.speed_dreams.SpeedDreams

File: org.speed.dreams.yaml [hunk 18]
```diff
@@ -0,0 +1,149 @@
+app-id: org.speed_dreams.SpeedDreams
```
Comment (TingPing) [inline org.speed.dreams.yaml]: The file name is wrong, it should be `org.speed_dreams.SpeedDreams.yml`

**Comment by TingPing** [comment]
bot, build org.speed_dreams.SpeedDreams

**Comment by TingPing** [comment]
Builds fails: https://flathub.org/builds/#/builders/5/builds/656

Looks like you never tested it locally..

**Comment by son-link** [comment]
Umm, i delete a file and download them from SourceForge. It will have been a temporary fall

**Comment by TingPing** [comment]
Sourceforge is garbage, use a mirror so it is avoided in the future because it will be down again.

**Comment by nedrichards** [comment]
bot, build org.speed_dreams.SpeedDreams

**Comment by nedrichards** [comment]
https://flathub.org/builds/#/builders/2/builds/5486

Doesn't build on arm and aarch64. If the project does not support them then add a flathub.json file to not build on those architectures like so: https://github.com/flathub/com.spotify.Client/blob/master/flathub.json

**Comment by leillo1975** [comment]
( https://github.com/flathub/flathub/pull/650/commits/4bb3d594de34396eace9d413d40e21c52e7c46cb )

Is it necessary to perform some other task to build the package?

**Comment by nedrichards** [comment]
bot, build org.speed_dreams.SpeedDreams

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.speed_dreams.SpeedDreams

@madlad, should someone besides you (and @son-link) get push access to the new repository?

**Comment by madbad** [comment]
Non sure if the last message was for me (I think you misspelled my nick).
Anyway I will ask on our mailing list if someone wants to get added

**Comment by barthalion** [comment]
Woop, yeah, sorry.

**Comment by leillo1975** [comment]
@madbad , I think that if someone deserves access here, that's you

**Comment by AsciiWolf** [comment]
It seems that Speed Dreams is missing on Flathub.org. It's showing in GNOME Software, but there is `(null)` instead of the `<ul>` list.

This is what `appstream-util validate` returns:
```
org.speed_dreams.SpeedDreams.appdata.xml: FAILED:
• tag-missing           : <translation> not specified
• style-invalid         : <p> requires sentence case [(null)]
• style-invalid         : <p> does not end in '.|:|!' [(null)]
• style-invalid         : <p> is too short [(null)]
Validation of files failed
```

**Comment by leillo1975** [comment]
> A question from someone who has no idea how all this is going: Is there a time frame or something missing for it to be published at https://flathub.org/home

Changes made:

https://github.com/flathub/org.speed_dreams.SpeedDreams/issues/1#issuecomment-433612809

**Pull request closed without merging**